From oliver@neukum.org  Thu Feb  5 09:21:00 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 5 Feb 2009 16:54:25 +0100
Subject: USB: serial: introduce a flag into the usb serial layer to tell drivers that their URBs are killed due to suspension
To: Greg KH <greg@kroah.com>, linux-usb@vger.kernel.org
Message-ID: <200902051654.26266.oliver@neukum.org>
Content-Disposition: inline


This patch introduces a flag into the usb serial layer to tell drivers
that their URBs are killed due to suspension. That is necessary to let
drivers know whether they should report an error back.

Signed-off-by: Oliver Neukum <oneukum@suse.de>

Hi Greg,

this is for 2.6.30. Patches to use this in drivers are under development.

	Regards
		Oliver

---

---
 drivers/usb/serial/usb-serial.c |    4 ++++
 include/linux/usb/serial.h      |    3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/usb-serial.c
+++ b/drivers/usb/serial/usb-serial.c
@@ -1067,6 +1067,8 @@ int usb_serial_suspend(struct usb_interf
 	struct usb_serial_port *port;
 	int i, r = 0;
 
+	serial->suspending = 1;
+
 	for (i = 0; i < serial->num_ports; ++i) {
 		port = serial->port[i];
 		if (port)
@@ -1084,8 +1086,10 @@ int usb_serial_resume(struct usb_interfa
 {
 	struct usb_serial *serial = usb_get_intfdata(intf);
 
+	serial->suspending = 0;
 	if (serial->type->resume)
 		return serial->type->resume(serial);
+
 	return 0;
 }
 EXPORT_SYMBOL(usb_serial_resume);
--- a/include/linux/usb/serial.h
+++ b/include/linux/usb/serial.h
@@ -130,7 +130,8 @@ struct usb_serial {
 	struct usb_device		*dev;
 	struct usb_serial_driver	*type;
 	struct usb_interface		*interface;
-	unsigned char			disconnected;
+	unsigned char			disconnected:1;
+	unsigned char			suspending:1;
 	unsigned char			minor;
 	unsigned char			num_ports;
 	unsigned char			num_port_pointers;
